Inconsistencies in generated files

### What happened?
Each time gqlgen runs it almost randomly picks whether it will use `interface{}` or `any` in some generated code.  I know they function the same, but is there a way to make the generated output more consistent?  

I don't think my use case is specifically important, I assume having a consistent output when generating code would be good regardless.  But the use case I have and how I'm reproducing the issue is with an automated test intended to catch when people change `.graphql` schema files and forget to regenerate the application code.  I have the following test workflow:
```
name: GQLGen Test

on:
  workflow_call: # Allows this to be executed by other workflows
  workflow_dispatch: # Allows this to be manually executed on any branch
  pull_request:
    branches: [ development ]

jobs:
  test:
    name: Generate graph
    runs-on: ubuntu-latest
    steps:

    - uses: actions/checkout@v4

    - name: Set up Go
      uses: actions/setup-go@v4
      with:
        go-version: 1.23.1

    - name: Run test gqlgen
      run: |
        go get -d github.com/99designs/gqlgen@v0.17.53
        cd app/api/graphql; go run github.com/99designs/gqlgen

    - name: Check file diff
      run: |
        if [ -n "$(git status --porcelain)" ]; then 
          echo "::error::The generated graphql resolvers do not match with those in the branch!"
          git status -v -v
          exit 1
        fi
```

The test failed as the generated files had the following diff:
```
Error: The generated graphql resolvers do not match with those in the branch!
On branch development
Your branch is up to date with 'origin/development'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   app/api/graphql/generated/graph/schema_gen.go

--------------------------------------------------
Changes not staged for commit:
diff --git i/app/api/graphql/generated/graph/schema_gen.go w/app/api/graphql/generated/graph/schema_gen.go
index f9ceeff..50af593 100644
--- i/app/api/graphql/generated/graph/schema_gen.go
+++ w/app/api/graphql/generated/graph/schema_gen.go
@@ -[23](https://github.com/orijinplus/backend/actions/runs/12228380133/job/34106552653#step:5:24)6836,12 +236836,12 @@ func (ec *executionContext) marshalNAnalyticsResult2ᚖorijinplusᚋappᚋapiᚋ
 	return ec._AnalyticsResult(ctx, sel, v)
 }
 
-func (ec *executionContext) unmarshalNAny2interface(ctx context.Context, v interface{}) (any, error) {
+func (ec *executionContext) unmarshalNAny2interface(ctx context.Context, v interface{}) (interface{}, error) {
 	res, err := graphql.UnmarshalAny(v)
 	return res, graphql.ErrorOnPath(ctx, err)
 }
 
-func (ec *executionContext) marshalNAny2interface(ctx context.Context, sel ast.SelectionSet, v any) graphql.Marshaler {
+func (ec *executionContext) marshalNAny2interface(ctx context.Context, sel ast.SelectionSet, v interface{}) graphql.Marshaler {
 	if v == nil {
 		if !graphql.HasFieldError(ctx, graphql.GetFieldContext(ctx)) {
 			ec.Errorf(ctx, "the requested element is null which the schema does not allow")
no changes added to commit (use "git add" and/or "git commit -a")
Error: Process completed with exit code 1.
```

So I pushed exactly that change and nothing else:
![image](https://github.com/user-attachments/assets/2f212595-f4cd-48e8-b063-4edf9afb0179)

And ran the test again but got the exact opposite from gqlgen:
```
Error: The generated graphql resolvers do not match with those in the branch!
On branch gqlgen-test
Your branch is up to date with 'origin/gqlgen-test'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   app/api/graphql/generated/graph/schema_gen.go

--------------------------------------------------
Changes not staged for commit:
diff --git i/app/api/graphql/generated/graph/schema_gen.go w/app/api/graphql/generated/graph/schema_gen.go
index 50af593..f9ceeff 100644
--- i/app/api/graphql/generated/graph/schema_gen.go
+++ w/app/api/graphql/generated/graph/schema_gen.go
@@ -[23](https://github.com/orijinplus/backend/actions/runs/12228411305/job/34106627491#step:5:24)6836,12 +236836,12 @@ func (ec *executionContext) marshalNAnalyticsResult2ᚖorijinplusᚋappᚋapiᚋ
 	return ec._AnalyticsResult(ctx, sel, v)
 }
 
-func (ec *executionContext) unmarshalNAny2interface(ctx context.Context, v interface{}) (interface{}, error) {
+func (ec *executionContext) unmarshalNAny2interface(ctx context.Context, v interface{}) (any, error) {
 	res, err := graphql.UnmarshalAny(v)
 	return res, graphql.ErrorOnPath(ctx, err)
 }
 
-func (ec *executionContext) marshalNAny2interface(ctx context.Context, sel ast.SelectionSet, v interface{}) graphql.Marshaler {
+func (ec *executionContext) marshalNAny2interface(ctx context.Context, sel ast.SelectionSet, v any) graphql.Marshaler {
 	if v == nil {
 		if !graphql.HasFieldError(ctx, graphql.GetFieldContext(ctx)) {
 			ec.Errorf(ctx, "the requested element is null which the schema does not allow")
no changes added to commit (use "git add" and/or "git commit -a")
Error: Process completed with exit code 1.
```

### What did you expect?
There should be no diff in the generated files if no changes are made to the source files they're generated from.

### versions
 - `go run github.com/99designs/gqlgen version 0.17.53`
 - `go version 1.23.1`

--- a/codegen/templates/templates.go
+++ b/codegen/templates/templates.go
@@ -248,7 +248,13 @@ func isDelimiter(c rune) bool {
 }
 
 func ref(p types.Type) string {
-	return CurrentImports.LookupType(p)
+	typeString := CurrentImports.LookupType(p)
+	// TODO(steve): figure out why this is needed
+	// otherwise inconsistent sometimes
+	if typeString == "interface{}" {
+		return "any"
+	}
+	return typeString
 }
 
 func obj(obj types.Object) string {

